tokio/src/sync/rwlock.rs

```diff
@@ -350,11 +120,13 @@ unsafe impl<T> Sync for RwLock<T> where T: ?Sized + Send + Sync {}
 unsafe impl<T> Send for RwLockReadGuard<'_, T> where T: ?Sized + Sync {}
 unsafe impl<T> Sync for RwLockReadGuard<'_, T> where T: ?Sized + Send + Sync {}
 unsafe impl<T> Sync for RwLockWriteGuard<'_, T> where T: ?Sized + Send + Sync {}
+unsafe impl<T> Sync for RwLockMappedWriteGuard<'_, T> where T: ?Sized + Send + Sync {}
 // Safety: Stores a raw pointer to `T`, so if `T` is `Sync`, the lock guard over
 // `T` is `Send` - but since this is also provides mutable access, we need to
 // make sure that `T` is `Send` since its value can be sent across thread
 // boundaries.
 unsafe impl<T> Send for RwLockWriteGuard<'_, T> where T: ?Sized + Send + Sync {}
+unsafe impl<T> Send for RwLockMappedWriteGuard<'_, T> where T: ?Sized + Send + Sync {}
```

Comment:
Should these impls be placed in the types' dedicated files? It might be more consistent as that's where all the other impls are.

Reply:
I guess that make sense.

As an aside, looking at them again, they appear to be overly strict. An `RwLockReadGuard` is like an immutable reference, so it can be `Sync` if `T: Sync`. Similarly for `RwLockWriteGuard` which can be `Send` as long as `T: Send`, and `Sync` as long as `T: Sync`.